cas-server-core-api-services/src/main/java/org/jasig/cas/services/RegisteredServiceAuthenticationPolicy.java

```diff
@@ -0,0 +1,44 @@
+package org.jasig.cas.services;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * This is {@link RegisteredServiceAuthenticationPolicy} that describes how a service
+ * should handle authentication requests.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public interface RegisteredServiceAuthenticationPolicy extends Serializable {
```

Comment:
@frett We should also focus on adding this sort of functionality to the management interface, of course, once the design here finalizes.